geth/node/manager.go

```diff
@@ -550,6 +575,25 @@ func (m *NodeManager) RPCClient() (*rpc.Client, error) {
 	return m.rpcClient, nil
 }
 
+// RPCClient exposes reference to RPC client connected to the running node.
+func (m *NodeManager) RPCClient() (*rpc.Client, error) {
```

Comment:
As I mentioned above, I am not sure we need this one. It might introduce some confusion which method use and can also result in bugs that will be hard to figure out. Caller should be 100% sure which client it needs to use.

Reply:
Am in agreement with this, but lets push it into a refactor issue, where we can deal with the implications of removal.